pkg/storage/detected/fields.go

```diff
@@ -0,0 +1,41 @@
+package detected
+
+import (
+	"github.com/grafana/loki/v3/pkg/logproto"
+)
+
+func MergeFields(fields []*logproto.DetectedField, limit uint32) []*logproto.DetectedField {
```

Comment:
this is not yet used right?

Reply:
It should be, so maybe I have a bug? It should be used by the query frontend when merging subqueries

Reply:
ah no, my bad.. its called directly in codec.go.. 👍